packages/itwinui-react/src/core/Dialog/Dialog.tsx

```diff
@@ -13,20 +13,29 @@ import { DialogButtonBar } from './DialogButtonBar.js';
 import { DialogMain } from './DialogMain.js';
 import { useMergedRefs, Box, Portal } from '../utils/index.js';
 import type { PolymorphicForwardRefComponent } from '../utils/index.js';
+import { Transition } from 'react-transition-group';
 
 type DialogProps = {
   /**
    * Dialog content.
    */
   children: React.ReactNode;
+  /**
+   * By default, the dialog/modal wrapper is always rendered regardless of `isOpen`.
+   *
+   * Pass `false` to avoid rendering the dialog/modal wrapper when `isOpen=false`.
+   *
+   * @default true
+   */
+  renderWrapperWhenClosed?: boolean;
```

Comment:
Do we need this prop? If there is no downside to always removing it from the DOM (e.g. all animations, etc work the same), then we don't need to make it configurable.

Reply:
Yes, all animations and the general behavior is unchanged. I thought of having a prop for this since this removal from the DOM behavior may be a breaking change.

But if we deem it not to be a breaking change, I can remove this prop and always remove the Dialog/Modal from the DOM when `isOpen=false`.

Reply:
Yeah, lets remove it then. It's not a breaking change, more of a bug fix. It never should have been kept in the DOM in the first place (it was only for animation reasons).

Reply:
Sounds good, removed (https://github.com/iTwin/iTwinUI/pull/1865/commits/b23fe980bd2dc29808d1276361fcde062a76740d).